### PR TITLE
docs: add "Error Handling" in "Upgrading from 3.x to 4.x"

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
From https://forum.codeigniter.com/showthread.php?tid=88993

- add "Error Handling" in "Upgrading from 3.x to 4.x"

See
- https://github.com/bcit-ci/CodeIgniter/blob/develop/system/core/Common.php#L600
- https://github.com/codeigniter4/CodeIgniter4/blob/develop/system/Debug/Exceptions.php#L202

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 1]
```diff
@@ -165,6 +165,23 @@ Hooks
 - The hook point ``post_system`` has moved just before sending the final rendered
   page.
 
+Error Handling
+==============
+
+- The behavior in CI4 has been slightly changed.
```
Comment (paulbalandan) [inline user_guide_src/source/installation/upgrade_4xx.rst:171]: At first glance, I don't quite understand it well. Can we make it in tabular form?

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 2]
```diff
@@ -165,6 +165,23 @@ Hooks
...
+    but other errors are logged (but depending on the ``log_threshold`` setting,
+    they may not be written to the log file). Only errors with the error level
+    ``E_ERROR | E_PARSE | E_COMPILE_ERROR | E_CORE_ERROR | E_USER_ERROR``
+    will stop the framework processing, regardless of the error level set in
+    ``error_reporting()``.
+  - In CI4, only errors with the error level set by ``error_reporting()`` in
+    **app/Config/Boot/{environment}.php** are logged (but depending on the
+    ``Config\Logger::$threshold`` setting, they may not be written to the log file).
+    All errors that are not ignored by ``error_reporting()`` will stop the
+    framework processing.
```
Comment (michalsn) [inline user_guide_src/source/installation/upgrade_4xx.rst:183]:
```suggestion
  - In CI3 the behavior was set in the **index.php** file:
      - errors ignored by ``error_reporting()`` was not logged;
      - other errors were logged (depending on the settings in ``log_threshold``, they may not have been written to the log file);
      - errors with an error level of ``E_ERROR | E_PARSE | E_COMPILE_ERROR | E_CORE_ERROR | E_USER_ERROR`` stopped framework processing, regardless of the error level set in ``error_reporting()``;
  - In CI4, the behavior is set in the **app/Config/Boot/{environment}.php** file:
      - errors with the error level set by ``error_reporting()`` are logged (but depending on the
    ``Config\Logger::$threshold`` setting, they may not be written to the log file);
      - all errors that are not ignored by ``error_reporting()`` will stop the framework processing;
```

Or maybe we can create a list, to make it more readable?

**Review by michalsn — APPROVED** [review]

**Comment by kenjis** [comment]
The production setting is `error_reporting(E_ALL & ~E_NOTICE & ~E_DEPRECATED & ~E_STRICT & ~E_USER_NOTICE & ~E_USER_DEPRECATED)`. So if `E_WARNING` error happens, in CI3 it is logged, but does not stop the framework. But in CI4, it is logged and stops the framework.

After all, this seems okay.
Screenshot 2023-12-22 17 02 30

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 3]
```diff
@@ -165,6 +165,28 @@ Hooks
 - The hook point ``post_system`` has moved just before sending the final rendered
   page.
 
+Error Handling
+==============
+
+- The behavior in CI4 has been slightly changed.
+
+  - In CI3 the behavior was set in the **index.php** file:
```
Comment (michalsn) [inline user_guide_src/source/installation/upgrade_4xx.rst:173]:
```suggestion
  - In CI3 the behavior is set in the **index.php** file:
```

**Review by michalsn — APPROVED** [review]

**Pull request merged**